pkg/ccl/backupccl/restore_job.go

```diff
@@ -355,31 +423,45 @@ func restore(
 		// to progCh.
 		for progress := range progCh {
 			mu.Lock()
-			var progDetails backuppb.RestoreProgress
-			if err := pbtypes.UnmarshalAny(&progress.ProgressDetails, &progDetails); err != nil {
-				log.Errorf(ctx, "unable to unmarshal restore progress details: %+v", err)
-			}
+			if clusterversion.ByKey(clusterversion.V23_1Start).LessEq(job.Payload().CreationClusterVersion) {
+				var progDetails backuppb.RestoreProgress
+				if err := pbtypes.UnmarshalAny(&progress.ProgressDetails, &progDetails); err != nil {
```

Comment:
nit: 1) declare and unmarshall prog details; 2) move the `mu.res.Add` call outside of the branching on the version gate to remove repeat code.

Reply:
Are you saying to do 1 in the same line or are you saying also to move it outside of the version branching?

Reply:
The later, move it all outside the version branching, as it's independent of the branching.